ts/src/uniswap.ts

```diff
@@ -0,0 +1,793 @@
+    //     const result = {};
+    //     for (let i = 0; i < currencies.length; i++) {
+    //         const currency = currencies[i];
+    //         const code = this.commonCurrencyCode (this.safeString (currency, 'symbol'));
+    //         const precision = this.parseNumber (this.parsePrecision (this.safeString (currency, 'decimals')));
+    //         result[code] = {
+    //             'id': this.safeString (currency, 'id'),
+    //             'code': code,
+    //             'info': currency,
+    //             'name': this.safeString (currency, 'name'),
+    //             'active': undefined,
+    //             'deposit': undefined,
+    //             'withdraw': undefined,
+    //             'fee': undefined,
+    //             'precision': precision,
+    //             'limits': {
+    //                 'amount': {
+    //                     'min': precision,
+    //                     'max': undefined,
+    //                 },
+    //                 'withdraw': {
+    //                     'min': undefined,
+    //                     'max': undefined,
+    //                 },
+    //             },
+    //         };
+    //     }
+    //     return result;
+    // }
+
+    async fetchMarkets (params = {}) {
+        /**
+         * @method
+         * @name binance#fetchMarkets
+         * @description retrieves data on all markets for binance
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[object]} an array of objects representing market data
+         */
+        const query = `query fetchMarkets($limit: Int, $offset: Int) {
+            pools(first: $limit, skip: $offset, orderBy: totalValueLockedUSD, orderDirection: desc) {
+              id
+              token0 {
+                id
+                symbol
+              }
+              token1 {
+                id
+                symbol
+              }
+              feeTier
+              feesUSD
+              txCount
+              liquidity
+              sqrtPrice
+              volumeUSD
+              volumeToken0
+              volumeToken1
+              observationIndex
+              collectedFeesUSD
+              collectedFeesToken0
+              collectedFeesToken1
+              totalValueLockedETH
+              totalValueLockedUSD
+              createdAtTimestamp
+              createdAtBlockNumber
+              feeGrowthGlobal0X128
+              feeGrowthGlobal1X128
+              totalValueLockedToken0
+              totalValueLockedToken1
+              liquidityProviderCount
+              totalValueLockedUSDUntracked
+            }
+          }`;
+        const request = {
+            'query': query,
+            'variables': {
+                'limit': 1000,
+                'offset': 0,
+            },
+        };
+        let markets = [];
+        while (true) {
+            const response = await this.uniswapPostUniswapV3 (this.extend (request, params));
```

Comment:
Are we going to add their v2 exchange? Because there are couple of forks of v2 exchanges.

Reply:
To start my plan was to focus on v3. I did a quick search for forks, but couldn't find any. Are there any good ones that might help with some of the code already done?